daemon/containerd/image_push.go

```diff
@@ -169,7 +169,14 @@ func (i *ImageService) pushRef(ctx context.Context, targetRef reference.Named, p
 	}
 
 	err = remotes.PushContent(ctx, pusher, target, store, limiter, platforms.All, handlerWrapper)
-	if err != nil && containerdimages.IsIndexType(target.MediaType) {
+	if err != nil {
+		if !containerdimages.IsIndexType(target.MediaType) {
+			if cerrdefs.IsNotFound(err) {
+				return errdefs.NotFound(err)
+			}
+			return errdefs.System(err)
```

Comment:
Maybe a comment that we consider any other error a system error (even if containerd may return a typed error) 🤔

Reply:
Isn't it just reiterating what the code does? 😅 
Maybe we shouldn't wrap it into System though.. Would you prefer to just return an unwrapped err?